public/components/custom_panels/custom_panel_view.tsx

```diff
@@ -9,37 +9,342 @@
+  };
+
+  // Fetch the custom panel on Initial Mount
   useEffect(() => {
-    fecthCustomPanel();
-    chrome.setBreadcrumbs([...parentBreadcrumb, {text: panelId, href: `/:${panelId}`}]); 
+    fetchCustomPanel();
   }, []);
 
-  return <div></div>;
+  // Check Validity of Time
+  useEffect(() => {
+    if (convertDateTime(end, false) < convertDateTime(start)) {
+      setToast('Invalid Time Interval', 'danger');
+      return;
+    }
+  }, [start, end]);
+
+  // Toggle input type (disabled or not disabled)
+  // Disabled when there no visualizations in panels or when the panel is in edit mode
+  useEffect(() => {
+    checkDisabledInputs();
+  }, [panelVisualizations, editMode]);
+
+  // Edit the breadcurmb when panel name changes
+  useEffect(() => {
+    chrome.setBreadcrumbs([
+      ...parentBreadcrumb,
+      { text: openPanelName, href: `${_.last(parentBreadcrumb).href}${panelId}` },
+    ]);
+  }, [openPanelName]);
+
+  return (
+    <div>
+      <EuiPage>
+        <EuiPageBody component="div">
+          <EuiPageHeader>
+            <EuiPageHeaderSection>
+              <EuiTitle size="l">
+                <h1>{openPanelName}</h1>
+              </EuiTitle>
+              <EuiFlexItem>
+                <EuiSpacer size="s" />
+              </EuiFlexItem>
+              Created on {panelCreatedTime}
+            </EuiPageHeaderSection>
+            <EuiPageHeaderSection>
+              <EuiFlexGroup gutterSize="s">
+                <EuiFlexItem>
+                  <EuiButton color="danger" onClick={deletePanel}>
+                    Delete
+                  </EuiButton>
+                </EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiButton onClick={renamePanel}>Rename</EuiButton>
+                </EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiButton>Export</EuiButton>
+                </EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiButton
+                    iconType={editMode ? 'save' : 'pencil'}
+                    onClick={editPanel}
+                    disabled={editDisabled}
+                  >
+                    {editMode ? 'Save' : 'Edit'}
+                  </EuiButton>
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            </EuiPageHeaderSection>
+          </EuiPageHeader>
+          <EuiPageContentBody>
+            <EuiFlexGroup gutterSize="s">
+              <EuiFlexItem>
+                <EuiFieldText
+                  placeholder="Use PPL to query log indices below. Use “source=” to quickly switch to a different index."
+                  value={filterBarValue}
+                  fullWidth={true}
+                  onChange={(e) => onChange(e)}
+                  disabled={inputDisabled}
+                />
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiSuperDatePicker
+                  isLoading={isLoading}
+                  dateFormat="MM/DD/YYYY hh:mm:ss A"
```

Comment:
could this be a constant?

Reply:
Will add it shared it to shared constants, so other modules can use it.